### ISSUE TITLE
clear "associated" with friend file transfer history from Transferred Files tab when one clears chat history of a friend

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#3219]
##### Brief Description

OS: Windows 7
qTox version: v1.4.0-2-g24a7f4d.
toxcore:  0.0.0
Qt:  5.5.1
…

Reproducible: Always
##### Steps to reproduce
1. Upload a "secret.2016.report.docx" to boss via qTox.
##### Observed Behavior

After successful transfer, I noticed qTox won't let me to remove "secret.2016.report.docx"
from "File Transfers" window - Uploads.
Boss said qTox won't let him to remove it from "File Transfers" - Downloads.
What's the point of keeping "file transfer history" anyway?
##### Expected Behavior

If I clear the chat(Left-click > Clear displayed messages) or close chat window AFTER download
or upload, qTox should forget about it.
No messengers log user's upload or download file history.
If qTox is secure messaging app, this is important.

**Comment by ghost** [comment]
Some people, aka paranoid thinks, qTox is collecting their file.
They claim it because qTox's "File Transfer" can't be cleared at all, and there's no evidence qTox is encrypting files before sent to target user.

**Comment by Chiitoo** [comment]
The list is cleared at least when the client is terminated (or re-started).  Whether or not that's intended behaviour though, I do not know a this time.

I can imagine plenty of users wanting to keep a list (a persistent list even) of the files transferred, but sure, it's not for everyone.  :]

Also, do keep in mind the early stages Tox and qTox are in.  That is, at least I would not expect anyone to fully trust these pieces of software just yet, especially not for security sensitive work.  I would hope “qTox is secure messaging app”, but there are no doubt issues we don't know of yet, which makes it more like “qTox aims to be a secure messaging app”.

That's how I personally feel about it.

I also believe this statement to be false: “No messengers log user's upload or download file history.”  I'm almost certain some do, but I don't believe it's important enough to prove for me to go finding and listing them.  Just saying.  ^^

**Comment by ghost** [comment]
> I also believe this statement to be false: “No messengers log user's upload or download file history.” 
> I'm almost certain some do, but I don't believe it's important

IIRC, Skype, Windows Messenger, Live Messenger, Pidgin does NOT log file transfer(on old version).

> I can imagine plenty of users wanting to keep a list (a persistent list even) of the files transferred

And I'm also sure plenty of privacy-aware users who use Tox like me do NOT want to keep
any history or log.

If developers can't agree on this, just add a single option to settings:

[X] Don't log anything about file transfers in "File Transfers".
[X] Clear "File Transfers" history after successful transfer.

and let people decide about it.

**Comment by verycrypt** [comment]
> I also believe this statement to be false: “No messengers log user's upload or download file history.” 
> I'm almost certain some do, but I don't believe it's important

IIRC, Skype, Windows Messenger, Live Messenger, Pidgin does NOT log file transfer(on old version).

> I can imagine plenty of users wanting to keep a list (a persistent list even) of the files transferred

And I'm also sure plenty of privacy-aware users who use Tox like me do NOT want to keep
any history or log.

If developers can't agree on this, just add a single option to settings:

[X] Don't log anything about file transfers in "File Transfers".
[X] Clear "File Transfers" history after successful transfer.

and let people decide about it.

**Comment by ghost** [comment]
> The list is cleared at least when the client is terminated (or re-started).

Yes this works, but needs improvement.
Like adding "Clear" button in "File Transfer" window.

**Comment by zetok** [comment]
> What's the point of keeping "file transfer history" anyway?

To know if you've e.g. send that document to someone, or not. Currently it's not saved in DB, but in the future it will be.

> If I clear the chat(Left-click > Clear displayed messages) or close chat window AFTER download
> or upload, qTox should forget about it.

Right.

> Some people, aka paranoid thinks, qTox is collecting their file.

Their paranoia is their problem. Not a qTox issue. I'd suggest them to educate themselves, since otherwise they're wasting their time fighting their own shadows.

> They claim it because qTox's "File Transfer" can't be cleared at all, and there's no evidence qTox is encrypting files before sent to target user.

Yes, qTox doesn't encrypt it before sending. Bytes are encrypted when sending. As for bullshit claims, look above.

> If qTox is secure messaging app, this is important.

Err, no. You didn't provide anything that would point to this conclusion. And it's not important for "security" reasons.

If you'd argue that it's important to have it because uneducated people perceive something as more "secure" if it offers a way to not save/clear metadata, then it might be a valid reasoning.

> Yes this works, but needs improvement.
> Like adding "Clear" button in "File Transfer" window.

Maybe.

# 

To sum it up:

What is needed:
- clear "associated" with friend file transfer history from `Transferred Files` tab when one clears chat history of a friend

In case where there would be a good (re-)design of `Transferred Files` tab, perhaps add a button to clear all file transfers.

**Comment by sudden6** [comment]
I would expect the file transfers to behave exactly like the chat history, if chat history is enable file transfers are also logged, if the user clears the chat history, file transfers are also cleared.

**Comment by ghost** [comment]
> if the user clears the chat history, file transfers are also cleared.

+1!!

Yes, I need private option, like "Incognito"(Chrome), "InPrivate"(IE) thing.

**Comment by zetok** [comment]
Related: #2125
